src/zstream.c

```diff
@@ -87,9 +87,51 @@ size_t git_zstream_suggest_output_len(git_zstream *zstream)
 		return ZSTREAM_BUFFER_MIN_EXTRA;
 }
 
+int git_zstream_get_output_chunk(
+	void *out, size_t *out_len, git_zstream *zstream)
+{
+	size_t in_queued, in_used, out_queued;
+
+	/* set up input data */
+	zstream->z.next_in = (Bytef *)zstream->in;
+	zstream->z.avail_in = (uInt)zstream->in_len;
+
+	if ((size_t)zstream->z.avail_in != zstream->in_len) {
```

Comment:
Is this handling the case where `z.avail_in = (uInt) zstream->in_len` overflows? If so, a small comment might be worthwhile.

Reply:
Updated to add a comment and an improved test.